pulpcore/app/replica.py

```diff
@@ -196,6 +198,16 @@ def sync_params(self, repository, remote):
         """This method returns a dict that will be passed as kwargs to the sync task."""
         raise NotImplementedError("Each replicator must supply its own sync params.")
 
+    def requires_syncing(self, distro):
+        no_content_change_since = _read_content_change_since_timestamp(distro)
+        if no_content_change_since and self.server.last_replication:
+            if self.server.last_replication < self.server.pulp_last_updated:
+                return True
+            if self.server.last_replication > no_content_change_since:
+                return False
```

Comment:
If we 
```suggestion
            if self.server.last_replication > parse_datetime(no_content_change_since):
                return False
```
here, you probably don't need the `_read_content_change_since_timestamp` function.

Reply:
The function takes care of converting and reading `distro.get("no_content_change_since")`. I found it better to extract it as a separate function. The function can return `None` or `pased_date`, which properly fits our needs. Do you insist on this?

Reply:
No, I just thought we test it for None here anyway. And we don't even attempt the conversion if the other one was None.